src/compiler/file_watcher/file_watcher.ts

```diff
@@ -93,17 +88,39 @@ async function reloadJs(
         }
 
         actor
-            .reload_js(timestamp, chunkNumber, chunk, reloadedJs.length)
+            .reload_js(timestamp, chunkNumber, chunk, BigInt(reloadedJs.length))
             .catch((error) => {
                 if (process.env.AZLE_VERBOSE === 'true') {
                     console.error(error);
                 }
             });
 
-        chunkNumber += 1;
+        chunkNumber += 1n;
     }
 
     if (process.env.AZLE_VERBOSE === 'true') {
         console.info(`Finished uploading chunks`);
     }
+
+    writeFileSync(reloadedJsPath, reloadedJs);
+}
+
+async function createActorReloadJs(): Promise<ActorReloadJs> {
+    const agent = await createAuthenticatedAgent(whoami());
```

Comment:
whoami might not work for authentication if the user has an identity that isn't Secp256

Reply:
This has been addressed, please review the changes for this carefully. I have a few points to discuss on this that I'd like for you to address in a separate PR if we both feel good about them.